app/controllers/users/confirmations_controller.rb

```diff
@@ -0,0 +1,74 @@
+class Users::ConfirmationsController < Devise::ConfirmationsController
+  skip_before_action :authenticate_user!
+
+  # PUT /resource/confirmation
+  def update
+    with_unconfirmed_confirmable do
+      if @confirmable.has_no_password?
+        @confirmable.attempt_set_password(params[:user])
+        if @confirmable.valid? and passwords_match?
+          do_confirm
+        else
+          do_show
+          @confirmable.errors.clear #so that we wont render :new
+        end
+      else
+        @confirmable.errors.add(:email, :password_already_set)
+      end
+    end
+
+    if !@confirmable.errors.empty?
+      self.resource = @confirmable
+      render 'devise/confirmations/new'
+    end
+  end
+
+  # GET /resource/confirmation?confirmation_token=abcdef
+  def show
+    with_unconfirmed_confirmable do
+      if @confirmable.has_no_password?
+        do_show
+      else
+        do_confirm
+      end
+    end
+    unless @confirmable.errors.empty?
+      self.resource = @confirmable
+      render 'users/confirmations/new'
+    end
+  end
+
+  def pending
+  end
```

Comment:
Do we need to override this method? Is the original implementation from Devise throwing `NotImplemented` exception otherwise? 🤔

Reply:
@naliwajek This is the "pending confirmation page" that is displayed to the user. It basically says "Check your email". We tell devise to redirect the user after registration here: https://github.com/alphagov/govwifi-admin/pull/17/files#diff-107c009feef83ea12f774836bb3e6a4bR8